electrum/gui/qt/util.py

```diff
@@ -1503,6 +1503,31 @@ def resizeEvent(self, e):
         return super().resizeEvent(e)
 
 
+from electrum.util import EventListener
+
+class QtEventListener(EventListener):
+
+    qt_callback_signal = QtCore.pyqtSignal(tuple)
+
+    def register_callbacks(self):
+        self.qt_callback_signal.connect(self.on_qt_callback_signal)
+        EventListener.register_callbacks(self)
+
+    def unregister_callbacks(self):
+        self.qt_callback_signal.disconnect()
+        EventListener.unregister_callbacks(self)
+
+    def on_qt_callback_signal(self, args):
+        func = args[0]
+        return func(self, *args[1:])
+
+# decorator for members of the QtEventListener class
+def run_in_qt_thread(func):
+    def decorator(self, *args):
```

Comment:
```suggestion
    @wraps(func)
    def decorator(self, *args):
```
(and `from functools import wraps` at the top)

This makes it IDE-friendlier but also, for example, you are logging in `EventListener.register_callbacks`, and this changes that from:
```
I | util | registering callback <bound method run_in_qt_thread.<locals>.decorator of <electrum.gui.qt.main_window.ElectrumWindow object at 0x7f3b9c1a6d30>>
```
to:
```
I | util | registering callback <bound method ElectrumWindow.on_event_verified of <electrum.gui.qt.main_window.ElectrumWindow object at 0x7f27f2512a60>>
```